src/support/timefuncs.c

```diff
@@ -28,31 +28,32 @@
 extern "C" {
 #endif
 
-#if defined(_OS_WINDOWS_)
-static double floattime(void)
+int jl_gettimeofday(struct jl_timeval *jtv)
 {
-    struct timeb tstruct;
-
-    ftime(&tstruct);
-    return (double)tstruct.time + (double)tstruct.millitm/1.0e3;
-}
+#if defined(_OS_WINDOWS_)
+    struct __timeb64 tb;
+    errno_t code = _ftime64_s(&tb);
+    jtv->sec = tb.time;
+    jtv->usec = tb.millitm * 1000;
 #else
-static double tv2float(struct timeval *tv)
+    struct timeval now;
+    int code = gettimeofday(&now, NULL);
+    jtv->sec = now.tv_sec;
+    jtv->usec = now.tv_usec;
+#endif
+    return code;
+}
+
+static double jtv2float(struct jl_timeval *jtv)
 {
-    return (double)tv->tv_sec + (double)tv->tv_usec/1.0e6;
+    return (double)jtv->sec + (double)jtv->usec*1.0e-6;
```

Comment:
The `(double)` casts are redundant here.  Multiplying by `1e-6` will already promote to double.  (And you don't need the `.0`).